katharsis-core/src/main/java/io/katharsis/queryspec/QuerySpecResourceRepository.java

```diff
@@ -0,0 +1,78 @@
+package io.katharsis.queryspec;
+
+import java.io.Serializable;
+import java.util.Set;
+
+import io.katharsis.queryParams.QueryParams;
+import io.katharsis.resource.exception.ResourceNotFoundException;
+
+/**
+ * Base repository which is used to operate on the resources. Each resource should have a corresponding repository
+ * implementation.
+ *
+ * @param <T>  Type of an entity
+ * @param <ID> Type of Identifier of an entity
+ */
+public interface QuerySpecResourceRepository<T, ID extends Serializable>{
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00119)